src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/JavaCallableWrapperGenerator.cs

```diff
@@ -283,13 +282,18 @@ void AddConstructor (MethodDefinition ctor, TypeDefinition type, string outerTyp
 				}
 		}
 
-		static IEnumerable<MethodDefinition> GetBaseMethods (MethodDefinition method)
+		static MethodDefinition GetBaseRegisteredMethod (MethodDefinition method)
 		{
 			MethodDefinition bmethod;
 			while ((bmethod = method.GetBaseDefinition ()) != method) {
 				method = bmethod;
-				yield return method;
+
+				var attributes = method.GetCustomAttributes (typeof (RegisterAttribute));
```

Comment:
If this was a hotspot, it might still be an allocation hotspot. The problem here is that `GetCustomAttributes` returns an `IEnumerable<CustomAttribute>`, thus we'll have to box the enumerator.

Maybe a `HasCustomAttributes (Type)` method could squeeze out a bit more, which also avoids the `Any()` LINQ by unrolling it?

`method.CustomAttributes` returns a concrete collection with a struct enumerator, maybe we should take advantage of that!

Reply:
Ok, what is even worse here, is I think every call to `GetCustomAttributes` calls either `Any` or `FirstOrDefault`.

We should avoid all this `IEnumerable` stuff!

Reply:
I'm going to do these next set of `IEnumerable` changes in another PR.